cypress/support/commands.js

```diff
@@ -1,25 +1,22 @@
-// ***********************************************
-// This example commands.js shows you how to
-// create various custom commands and overwrite
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+Cypress.Commands.add("addMock", async (url, method, response) => {
```

Comment:
This is really nice and much easier than Mountebank. It looks especially suited to "fire and forget" scenarios where we want to mock out a single response. How does it cope if you need to provide different responses in order, e.g. a 403 for this response on the first request, then a 200 for the next one? Is there an implicit FIFO queue or something?

Reply:
I don't think we can specify an order of responses, but it can do request matching and we can insert/overwrite on-the-fly between test steps, which I think should cover most scenarios.